.jupyter/jupyterhub_config.py

```diff
@@ -232,6 +233,20 @@ def get_env(self):
 
     return env
 
+  async def get_url(self):
+    pods = oapi_client.resources.get(kind='Pod', api_version='v1')
+    pod = pods.get(name="jupyterhub-nb-%s" % escape(self.user.name), namespace=custom_notebook_namespace if custom_notebook_namespace else namespace)
+
+    url = None
+    try:
+        url = "http://%s:%d/user/%s/" % (pod["status"]["podIP"], pod['spec']['containers'][0]['ports'][0]['containerPort'], self.user.name)
```

Comment:
Should it be http://%s:%d/user/%s/**lab**? Or is it appended automatically by JH?
I would also use https, but that's a nitpick.

Reply:
We cannot use https - this is in cluster communication which is not encrypted
I am not sure, the only goal here is to give JH real URL that it can reach to verify the user pod is running - the /lab suffix is used on the route, so I believe it is not necessary here